source/common/thread_local/thread_local_impl.cc

```diff
@@ -66,17 +66,31 @@ ThreadLocalObjectSharedPtr InstanceImpl::SlotImpl::getWorker(uint32_t index) {
 
 ThreadLocalObjectSharedPtr InstanceImpl::SlotImpl::get() { return getWorker(index_); }
 
-void InstanceImpl::SlotImpl::runOnAllThreads(const UpdateCb& cb, Event::PostCb complete_cb) {
+Event::PostCb InstanceImpl::SlotImpl::dataCallback(const UpdateCb& cb) {
   // See the header file comments for still_alive_guard_ for why we capture index_.
-  parent_.runOnAllThreads(
-      wrapCallback([cb, index = index_]() { setThreadLocal(index, cb(getWorker(index))); }),
-      complete_cb);
+  return [still_alive_guard = std::weak_ptr<bool>(still_alive_guard_), cb, index = index_] {
+    if (still_alive_guard.expired()) {
+      return;
+    }
```

Comment:
I get why you duplicated this logic to avoid another lambda, but it's kind of confusing. At minimum can you comment why it's duplicated? Is there any better way we could avoid the duplication? Otherwise LGTM!

/wait

Reply:
A comment is reasonable. I thought about a helper method to reduce the duplicate code, but it seemed like the extra functional layer would not make anything clearer and it would cost lines of code. But I'll definitely add a comment.